cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -756,19 +754,21 @@ impl ExprOrSpecial<'_> {
                 }
             },
             Self::Var(var, _) => {
-                errs.push(err::ParseError::ToAST(format!(
+                errs.push(err::ParseError::UnexpectedNonLiteral(format!(
                     "not a string literal: {var}"
```

Comment:
this error message also won't work with the `#[error]` template on `UnexpectedNonLiteral`.  And more occurrences below

Reply:
Would love to see a few tests that explicitly compare the error message to some constant string, so that we can look at the constant strings and see they're reasonable, rather than piecing things together ourselves in our heads